benchmark/CMap.hs

```diff
@@ -9,29 +10,38 @@
 {-# OPTIONS_GHC -fplugin GHC.TypeLits.KnownNat.Solver #-}
 
 module CMap
-       ( benchMap
-       , prepareBenchMap
+       ( spec
        ) where
 
-import Criterion.Main (Benchmark, bench, bgroup, nf)
+import Criterion.Main (bench, nf, env, whnf)
 
 import Prelude hiding (lookup)
 
-import Control.DeepSeq (rnf)
-import Control.Exception
+import Common
 import Data.Maybe (fromJust)
 import Data.Proxy (Proxy (..))
 import Data.Typeable (Typeable)
 import GHC.TypeLits
 
-import Data.TypeRep.CMap (TypeRepMap (..), empty, insert, keys, lookup)
+import Data.TypeRep.CMap (TypeRepMap (..), empty, insert, lookup)
 
-benchMap :: Benchmark
-benchMap = bgroup "map"
-    [ bench "lookup"     $ nf tenLookups bigMap
-    --, bench "insert new" $ whnf (\x -> rknf $ insert x bigMap) (Proxy :: Proxy 9999999999)
-    --, bench "update old" $ whnf (\x -> rknf $ insert x bigMap) (Proxy :: Proxy 1)
-    ]
+spec :: BenchSpec
+spec = BenchSpec
+  { benchLookup = Just $ \name ->
+      env mkBigMap $ \ ~bigMap ->
+        bench name $ nf tenLookups bigMap
+  , benchInsertSmall = Just $ \name -> 
+      bench name $ whnf (inserts empty 10) (Proxy :: Proxy 99999)
```

Comment:
Is `whnf` enough here? Are there any reasons to not use `nf` always?

Reply:
our structures are `whnf` strict. `nf` will force additional traversal even structure is already forced, in case of the big maps this traversal may introduce a large performance hit.